### PR TITLE
Adds RPGs to Nuke Ops

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
add: Nuke Ops now have the ability to purchase a usable RPG, as well as a couple different types of rockets for it.
/:cl:

untitled-1

They look awesome, they **feel** awesome, and they're completely unused. It'll be _perfect_.

Also added them to the summon guns weapon pool. These babies need some love.

### CODE REVIEW

**Comment by Metroidlover** [comment]
Sounds fun and stupid, so perfect.

**Comment by iskyp** [comment]
wtf why does the picture have a reloadable rpg this pr is unusable

**Comment by evsey9** [comment]
Every day we become one step closer to becoming goon

**Comment by GuyonBroadway** [comment]
Good, goon knows how to have a good time.

**Comment by Potato-Masher** [comment]
RAWKET LAWNCHAIR

**Comment by Dorsisdwarf** [comment]
I'm not exactly a fan of a ranged insta-kill weapon that creates a breach and fucks up everyone around the target, even if its one-shot and expensive

**Comment by WJohn** [comment]
Thiiis might not be the smartest addition. The explosion this thing lets off is an instant kill on humans, even the HoS won't survive it. For 14 TC, this thing is pretty expensive for 1 use, but also simultaneously too cheap for what it could do.

MORE than that, though, is that isntakills are bad gameplay. Its high price however means that if you miss you just wasted a huge amount of TC which is detrimental to your team. It's super high risk for a really unfun way to be killed.

I just don't see this as being worth adding, and it's kind of redundant as far as super-high power items are concerned. The l6 saw kills people and mechs just as effectively but boasts a 50 round mag and loads of spam for only 4 TC more, and the m90 already has a grenade launcher function (albeit less powerful).

**Comment by XDTM** [comment]
Instakills are unfun, but when playing against ops they're pretty much part of the gamemode. Sniper rifles are similarly lethal and have less counterplay, and most other guns will kill you in a few shots anyway.

Plus, it has other uses beyond killing; it could be used to breach an exterior wall or a reinforced room without having to get close to plant bombs or C4. Thinking about it like a ranged bomb explains its all-or-nothing nature as well.

**Comment by WJohn** [comment]
Sniper rifles are also terrible and shouldn't be in the game. And the l6 saw probably needs some severe nerfs because it's frankly kind of ridiculously powerful.

As for breaching a door or wall, that is *insanely* wasteful given how a stick of C4 costs literally like 1 TC whereas this is ranged, can by used on people, instakills them, and costs 14 TC. Why on earth would you use it on anything other than a mech/well armored foe?

**Comment by ExcessiveUseOfCobblestone** [comment]
A large difference between the sniper and rocket is the AoE.

**Comment by davethwave** [comment]
Also the fact that you can reload a sniper...

**Comment by WJohn** [comment]
They both effectively fuck you over if you get hit. The sniper also knocks you down for long enough to follow up with another shot, and its armor piercing makes armor irrelevant. It's also very likely to take off a limb, and it has a mag that can be reloaded, too.

They're both terribly unfun and hideously powerful weapons for different reasons. No one should have them.

**Comment by Kagus** [comment]
That could make some sense, but most station areas are cramped enough that you can probably blast someone with the AoE anyways. But, hey, then they're not on ground zero, and the wall's probably not gonna be there for a second shot...

You'd have to do something about FPS strats and prevent the projectile from terminating on the floortile the person is standing on, though.

**Comment by ShizCalev** [comment]
Few notes on the AT launcher's current munition

- Single shot
- Can't fit in a bag or be put on back
- Requires both hands to be free to use (no rocket / shield combo)
- Will kill on a -direct hit- (as would be expected with a HEDP round.) https://gfycat.com/BrokenFemaleEuropeanpolecat
- Blast radius is 3 tiles (counting epicenter), with damage to mobs within the radius around the epicenter being between 30 and 50. https://gfycat.com/EagerFrailIndianspinyloach
- Chance of limb loss & knockdown in that radius.
- Flames reach out 4 tiles (including epicenter.)
- Can be dodged https://gfycat.com/RegalCourteousAtlasmoth

**Comment by XDTM** [comment]
Projectiles by default go on until they hit something, unless you target a mob (which is where we add the dodge code). They won't land on a tile, so that's not a problem.

**Comment by XDTM** [comment]
@ShizCalev the issue is not really that they're op but rather that they're too "all or nothing", where the "all" is unfun for the victim, and the "nothing" is a waste of a lot of TC for the shooter.
Making them a bit weaker by adding the dodge chance as well as increasing their consistency by allowing multiple shots the problem should be reduced.

**Comment by WJohn** [comment]
Honestly I don't really see the need for the compromise XDTM is suggesting. These don't have much of a reason to be in the game when we have loads of other explosive weapons and  the m90 and mechs already handle the explosive attacks without being utterly worthless if they miss or stupidly lethal if they hit.

**Comment by XDTM** [comment]
> without being utterly worthless if they miss or stupidly lethal if they hit.

I mean, that's the point of my suggestion, to avoid that _if_ RPGs are added. Having more than one shot means that one miss doesn't cost 14 TC, and the dodge chance (i mean like 90%) means that ops shouldn't try for direct hits in the first place, but rather they should try to hit walls near their targets for large but not instagib levels of damage, leading to a possible chance of survival if you're quick or armored.

As for why we should add this, i agree they're not particularly necessary, but they sure fit into the equipment of a paramilitary organization, and i think they might result in some cool strategies.

**Comment by ShizCalev** [comment]
Alright, updated. 

The launcher is now reusable. Additional rockets for it can be found in the uplink. It's now preloaded with the weaker SRM-8 rocket on purchase.

Resprited the rocket for the launcher, and also updated the SRM-8's sprite to have animated fire.

**Comment by imsxz** [comment]
to the people saying this is too OP: fireball/explosive lances

**Comment by WJohn** [comment]
those are dumb as well

just because we have stupid, op garbage in some parts of the game doesn't justify the addition of other stupid, op garbage. that is exactly what power creep is.

we'll have to see how these turn out now with the latest changes that made them less all-or-nothing.

**Comment by WarRage333** [comment]
Nice a good addition ruined into the ground because a total of two people cried about it

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -220,6 +220,15 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	surplus = 0
 	include_modes = list(/datum/game_mode/nuclear)
 
+/datum/uplink_item/dangerous/rawketlawnchair
+	name = "84mm Rocket Propelled Grenade Launcher"
+	desc = "A reusable rocket propelled grenade launcher preloaded with a low-yield 84mm SRM-8 HE round. \
```
Comment (Wilchenx) [inline code/modules/uplink/uplink_items.dm]: srm-8 stands for short range missle x 8
its not the actual name for a projectile, its a name for the weapon

SRM-20 would be a huge rack with 20 short range missles

https://mwo.gamepedia.com/Short_Range_Missile_6

Comment (ShizCalev) [reply code/modules/uplink/uplink_items.dm]: https://github.com/tgstation/tgstation/blob/1b50671ca072dd646e98bee971a6d20d5c06a2de/code/modules/projectiles/projectile/special/rocket.dm#L32-L33

?

Comment (Wilchenx) [reply code/modules/uplink/uplink_items.dm]: its not like you ever see projectiles anyway, it was probably named like that to avoid confusion

Comment (Wilchenx) [reply code/modules/uplink/uplink_items.dm]: renaming things to just SRM would be correct with the refrence of the items

http://www.sarna.net/wiki/Short_Range_Missile

**Comment by ishitbyabullet** [comment]
Atlaunchers/RPGs should be admin only. They're too OP or bullshit and take no effort to get. If I'm wrong then tell me one time an admin did a rocket launcher tc trade.

No, it doesn't count if it was a chameleon gun or the rocket launcher was empty therefore useless.

**Comment by ExcessiveUseOfCobblestone** [comment]
I thought this was for ops, not traitors. Ops don't typically tc trade.

File: code/modules/uplink/uplink_items.dm (hunk 1 above)
Comment (Dorsisdwarf) [reply code/modules/uplink/uplink_items.dm]: the projectile is referred to as the SRM-8 rocket because it gets fired from a SRM-8 and nothing else

**Comment by ishitbyabullet** [comment]
@ExcessiveUseOfCobblestone If traitors can't get it (without the admin making it basically a meme) and it can't be justified for a station job to get it (chef getting CQC) then what makes it fair for a nuke op to have it? You don't even get this weapon from wizard summon guns and it has a fucking pistol which shoots explosive projectiles. As nukie, you already have a wall piercing sniper rifle, l6 saw, dark mauler, and other stuff which I don't want to say otherwise god forbid. You don't need an instakill rocket launcher. 14 tc is literally cheaper than the fucking shielded hardsuit, sniper kit, cyborgs, mechs, and the syndicate balloon. An instakill weapon, even if you're likely to miss, should still cost **more than the fucking other op weapons the nuke ops have.**

But surely 14 tc is _balanced._ It's not like nuke ops can declare war and get a shit ton of TC so they could theoretically spam rockets easily.

Nah, that won't happen. Naaaaaaaaaaaaaaaaaah.

WJohnston is right in that we don't need fucking powercreep. If people are sad that they can't get a RPG then they should give an idea on how an admin can do ONE event with them besides team deathmatch search and destroy free for all with killstreaks.

**Comment by vuonojenmustaturska** [comment]
sure, seems fun, let's try it

**Comment by Wilchenx** [comment]
>its still named srm-8

i am so fucking triggered

**Comment by ExcessiveUseOfCobblestone** [comment]
@ishitbyabullet I wasn't disagreeing it's power level, I haven't seen it.

When you discussed TC trade I thought it involved traitors since, again, ops don't tc trade.

**Comment by ishitbyabullet** [comment]
I guess balance doesn't exist.

Must be for dumbasses, I guess.

**Pull request merged**